src/main/java/mate/acadamy/internetshop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,157 @@
+                shoppingCarts.add(getShoppingCartFromResultSet(resultSet));
+            }
+            return shoppingCarts;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting of list of shopping carts is failed", e);
+        }
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        deleteProductsFromCart(shoppingCart);
+        addProductsToCart(shoppingCart);
+        return shoppingCart;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM shopping_carts WHERE cart_id=?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            statement.executeUpdate();
+            return true;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Delete of shopping cart with id="
+                    + id + " is failed", e);
+        }
+    }
+
+    private void addProductsToCart(ShoppingCart shoppingCart) {
+        String query = "INSERT INTO "
+                + "internet_shop.shopping_carts_products(cart_id, product_id) "
+                + "VALUES(?,?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            for (Product product : shoppingCart.getProducts()) {
+                PreparedStatement statement = connection.prepareStatement(query);
+                statement.setLong(1, shoppingCart.getShoppingCartId());
+                statement.setLong(2, product.getProductId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add products to shopping cart", e);
+        }
+    }
+
+    private void deleteProductsFromCart(ShoppingCart shoppingCart) {
+        String query = "DELETE FROM internet_shop.shopping_carts_products WHERE cart_id=?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, shoppingCart.getShoppingCartId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete products from shopping cart", e);
+        }
+    }
+
+    public ShoppingCart getShoppingCartFromResultSet(ResultSet resultSet) throws SQLException {
+        Long cartId = resultSet.getLong("cart_id");
+        Long userId = resultSet.getLong("user_id");
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setShoppingCartId(cartId);
+        shoppingCart.setUserId(userId);
+        shoppingCart.setProducts(getProductsOfCart(cartId));
+        return shoppingCart;
+    }
+
+    private List<Product> getProductsOfCart(Long id) {
+        String query = "SELECT products.product_id, products.name, products.price "
```

Comment:
Можемо використати aliases?
```suggestion
        String query = "SELECT p.product_id, p.name, p.price "
```

Reply:
Yes, done